ext/mysqlnd/mysqlnd_ps_codec.c

```diff
@@ -241,7 +242,21 @@ ps_fetch_time(zval * zv, const MYSQLND_FIELD * const field, const unsigned int p
 		t.time_type = MYSQLND_TIMESTAMP_TIME;
 	}
 
-	length = mnd_sprintf(&value, 0, "%s%02u:%02u:%02u", (t.neg ? "-" : ""), t.hour, t.minute, t.second);
+    if (field->decimals > 0 && field->decimals < 7) {
+        length = mnd_sprintf(
+            &value,
+            0,
+            "%s%02u:%02u:%02u.%0*" ZEND_ULONG_FMT_SPEC,
+            (t.neg ? "-" : ""),
+            t.hour,
+            t.minute,
+            t.second,
+            field->decimals,
+           (int) (t.second_part / pow(10, 6 - field->decimals))
```

Comment:
For `int` the placeholder would be `%d` ;)

Reply:
Thanks. :D Sure enough. I should cast to UINT32_T, given that it can only take a value between 0-999999.
But that is theoretical only, as now I have the problem that my last commit failed the test, whilst it ran without failure or being skipped locally.
Apart from versions, etc, the only obvious difference is that I use "127.0.0.1" as MYSQL_TEST_HOST without using mysqld.sock (that seems to be the default for "localhost" for years now).

Reply:
As for the target branch.. most people (like me) are forced to use emulated_prepares with PDO (that is on by default anyway, but many of us would have reasons to turn it off).
Now having a delta of 1 second was totally acceptable like, um, 10 years ago, but in a HA system, many things could and should happen in one second.
There are some bug reports for stuff like PHP frameworks and so on, with people having no idea what and why is this happening.

Reply:
The support for actually working microseconds is in since MySQL 5.6.4 (milestone 7, released on 2011-12-20; stable as 5.6.10 since 2013-02-05; neither seems yesterday).